zrml/swaps/src/lib.rs

```diff
@@ -1571,11 +1571,8 @@ mod pallet {
                 let amount = T::AssetManager::free_balance(asset, &pool_account);
                 T::AssetManager::slash(asset, &pool_account, amount);
             }
-            let pool_share_id = Self::pool_shares_id(pool_id);
-            let (_, liquidity_providers) =
-                T::AssetManager::accounts_by_currency_id(pool_share_id).unwrap_or((0usize, vec![]));
-            let _ =
-                T::AssetManager::destroy_all(pool_share_id, liquidity_providers.iter().cloned());
```

Comment:
Double check from @maltekliemann : No negative consequences here? I think the storage migration is possible the same way as you @sea212 explained [here](https://github.com/zeitgeistpm/zeitgeist/issues/792#issuecomment-1262115357) .